lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -5742,7 +5861,14 @@ Error glow::optimizeFunction(Function *F, const Backend &B,
   // Allow the backend to transform the graph after lowering.
   RETURN_IF_EXPECTED_IS_ERR(B.transformPostLowering(F, cctx, devInfo));
 
-  if (!B.shouldPreQuantizeConstants()) {
+  // Convert unsupported INT8 node to FP16 or FP32 and add dequantization and
+  // quantization nodes at the input and output of the corresponding node.
+  if (cctx.precisionConfig.externalQuantization) {
+    runDCEPass(F, cctx);
+    ::glow::externalQuantizationPass(B, F, cctx);
+  }
```

Comment:
Why not just do this before the check for `shouldPreQuantizeConstants()`, and leave `shouldPreQuantizeConstants()` alone? Otherwise if we originally had `Constant -> Quantize -> Node`, and then `convertQuantizedConstants()` converts that to  `Constant' -> Node`, and then here we result in `Constant' -> Dequantize -> Node`, then we lose precision on the Constant. If we do it before then we'd have `Constant -> Quantize -> Dequantize -> Node` and then the `Quantize -> Dequantize` could be optimized away.

Reply:
Hey @jfix71, We have to do the compatibility dequantization pass after transformPostLowering pass only. Hence we have put it after shouldPreQuantizeConstants(). Maybe we can put a glow::optimize pass after compatibility dequantization and convertQuantizedConstants(). WDYT?